### PR TITLE
Increase test coverage to 83.8% and fix SetupHystrixPrometheus panic

### PR DESCRIPTION
**Pull request by ankurs**
## Summary
- Add comprehensive test suite (`core_coverage_test.go`) covering server lifecycle, HTTP handler routing, initializers, VTProto codec, and utility functions
- **Bug fix**: Wrap `SetupHystrixPrometheus` with `sync.Once` to prevent panic on duplicate Prometheus collector registration
- Coverage: 21.3% → 83.8%
- Regenerate READMEs via gomarkdoc

## Test plan
- [ ] `go test -race -cover ./...` passes with 83.8% coverage
- [ ] Verify `SetupHystrixPrometheus` no longer panics when called multiple times

## Summary by CodeRabbit

* **Documentation**
  * Updated API reference documentation with corrected line references for core package initializers
  * Revised service startup documentation to reflect current health check endpoints

* **Bug Fixes**
  * Fixed duplicate Prometheus collector registration panic that could occur on multiple initializations

* **Tests**
  * Added comprehensive test suite covering core package functionality and edge cases

### CODE REVIEW

File: core_coverage_test.go [hunk 1]
```diff
@@ -0,0 +1,888 @@
...
+	time.Sleep(300 * time.Millisecond)
+
+	if err := instance.Stop(2 * time.Second); err != nil {
+		t.Fatalf("Stop failed: %v", err)
+	}
+
+	err := <-errCh
+	if err != nil && err != http.ErrServerClosed && err.Error() != "grpc: the server has been stopped" {
+		t.Fatalf("unexpected Run error: %v", err)
+	}
```
[the remark this answers is not in the document]
Comment (ankurs) [reply core_coverage_test.go]: Fixed — now using `errors.Is(err, grpc.ErrServerStopped)` instead of string comparison.
